unit_tests/library/match.cpp

```diff
@@ -184,7 +184,7 @@ BOOST_AUTO_TEST_CASE(test_match_fb_serialization)
     const auto matchings = fb->routes();
     const auto &number_of_matchings = matchings->size();
 
-    for (const auto &waypoint : *waypoints)
+    for (const auto waypoint : *waypoints)
```

Comment:
Also clang-related?

Reply:
Yes, similar errors to above: https://github.com/mjjbell/osrm-backend/runs/2957246502?check_suite_focus=true#step:15:2784

> error: loop variable 'destination' is always a copy because the range of type 'const flatbuffers::Vector<flatbuffers::Offset<osrm::engine::api::fbresult::Waypoint> >' does not return a reference [-Werror,-Wrange-loop-analysis]
    for (const auto &destination : *destinations_array)
                     ^
/Users/runner/work/osrm-backend/osrm-backend/unit_tests/library/table.cpp:335:10: note: use non-reference type 'const osrm::engine::api::fbresult::Waypoint *'
    for (const auto &destination : *destinations_array)